### PR TITLE
Adds CI for SQL versions

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Adds CI for robustly checking the SQL version

## Why It's Good For The Game

Catches some stupid errors

## Images of changes

image
image
image
image

## Testing

See above

## Changelog
NPFC

### CODE REVIEW

**Review by AffectedArc07 — APPROVED** [review]
Well we have a check that makes sure code matches SQL itself. 

The lack of checking for SQL version based on update file is its own issue. That said, off me.

**Review by Burzah — APPROVED** [review]

File: tools/ci/verify_sql_version.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            print(f"::error file={filename},title=SQL Version mismatch::{filename}: {RED}{message}{NC}")
+        else:
+            print(f"::error title=SQL Version mismatch:: {RED}{message}{NC}")
+    else:
+        if filename != None:
+            print(f"{filename}: {RED}{message}{NC}")
+        else:
+            print(f"ERROR: {RED}{message}{NC}")
+
+def get_sql_folder_version():
```
Comment (warriorstar-orion) [inline tools/ci/verify_sql_version.py:28]: can you annotate the return types?

File: tools/ci/verify_sql_version.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        else:
+            print_error(f"No byond define for SQL found in {define_path}", define_path)
+            exit_code = 1
+    else:
+        print_error(f"File containing the byond define for the SQL version does not exist ({define_path})")
+        exit_code = 1
+
+    updates_file, updates_folder_sql = get_sql_folder_version()
+
+    if updates_folder_sql <= 0 or not isinstance(updates_folder_sql, int) or updates_file == None:
```
Comment (warriorstar-orion) [inline tools/ci/verify_sql_version.py:88]: `not updates_file` or `update_files is None`

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]

File: tools/ci/verify_sql_version.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        else:
+            print(f"::error title=SQL Version mismatch:: {RED}{message}{NC}")
+    else:
+        if filename != None:
+            print(f"{filename}: {RED}{message}{NC}")
+        else:
+            print(f"ERROR: {RED}{message}{NC}")
+
+def get_sql_folder_version():
+    folder_path = "SQL/updates"
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply tools/ci/verify_sql_version.py:29]: Because I'm clueless with python, updateed to use pathlib

**Review by warriorstar-orion — APPROVED** [review]
LGTM (Let's Guarantee These Migrations)

**Pull request merged**